geode-core/src/main/java/org/apache/geode/cache/RegionShortcut.java

```diff
@@ -192,30 +192,36 @@
    * {@link DataPolicy} to {@link DataPolicy#PERSISTENT_REPLICATE} and {@link EvictionAttributes}
    * are set to {@link EvictionAlgorithm#LRU_HEAP} with {@link EvictionAction#OVERFLOW_TO_DISK}.
    */
-  LOCAL_PERSISTENT_OVERFLOW,
+  LOCAL_PERSISTENT_OVERFLOW(DataPolicy.PERSISTENT_REPLICATE),
 
   /**
    * A PARTITION_PROXY has no local state and forwards all operations to a PARTITION or a
    * PARTITION_PERSISTENT that exists in its peers. The actual RegionAttributes for a
    * PARTITION_PROXY region set the {@link DataPolicy} to {@link DataPolicy#PARTITION} and the
    * local-max-memory to 0.
    */
-  PARTITION_PROXY,
+  PARTITION_PROXY(DataPolicy.PARTITION),
   /**
    * A PARTITION_PROXY_REDUNDANT has no local state and forwards all operations to a
    * PARTITION_REDUNDANT or a PARTITION_REDUNDANT_PERSISTENT that exists in its peers. The actual
    * RegionAttributes for a PARTITION_PROXY_REDUNDANT region set the {@link DataPolicy} to
    * {@link DataPolicy#PARTITION}, the local-max-memory to 0, and the redundant-copies to 1.
    */
-  PARTITION_PROXY_REDUNDANT,
+  PARTITION_PROXY_REDUNDANT(DataPolicy.PARTITION),
   /**
    * A REPLICATE_PROXY has no local state and forwards all operations (except queries) to a
    * REPLICATE or a REPLICATE_PERSISTENT that exists in its peers. Queries will be executed on this
    * PROXY region. The actual RegionAttributes for a REPLICATE_PROXY region set the
    * {@link DataPolicy} to {@link DataPolicy#EMPTY} and {@link Scope} to
    * {@link Scope#DISTRIBUTED_ACK}.
    */
-  REPLICATE_PROXY;
+  REPLICATE_PROXY(DataPolicy.EMPTY);
+
+  private DataPolicy dataPolicy;
```

Comment:
This is probably not a safe change for backwards compatibility.
RegionShortcut is an external class which means customers could be doing with it whatever they want.
It is also an "enum" which means it is Serializable (see java.lang.Enum). So I think by adding this private field you have caused its serialization to not be compatible with older versions. You could add a test that serializes a RegionShortcut with a previous version of geode and then tries to deserialize with this new version but I'm pretty sure it will be broken.
It would be safe to add a method (like getDataPolicy) but you would need to change it to only use state that exists on the old version of RegionShortcut. Which would mean using a switch statement and for each enum type returning the corresponding DataPolicy. I think it would look like:
switch (this) {
  case REPLICATE_PROXY: return DataPolicy.EMPTY;
...